app/controllers/api/v1/locations_controller.rb

```diff
@@ -3,16 +3,18 @@ class Api::V1::LocationsController < ApplicationController
 
   def create
     data = JSON.parse(request.body.read)
-    new_location = Location.create(
-      name: data["name"],
-      description: data["description"],
-      address: data["address"],
-      city: data["city"],
-      state: data["state"],
-      zip: data["zip"],
-      url: data["url"],
-      user_id: current_user.id)
-    render json: new_location
+    if current_user
+      new_location = Location.create(
+        name: data["name"],
+        description: data["description"],
+        address: data["address"],
+        city: data["city"],
+        state: data["state"],
+        zip: data["zip"],
+        url: data["url"],
+        user_id: current_user.id)
+      render json: new_location
+    end
```

Comment:
There should also be an `else` clause here that renders an error.

You may want to have this as an `authenticate_user` method that does `Location.new` and renders the errors in JSON if it fails to save.